src/c2goto/library/pthread_lib.c

```diff
@@ -141,7 +147,7 @@ __ESBMC_HIDE:;
     __ESBMC_thread_key *l = search_key(i);
     if (__ESBMC_thread_key_destructors[i] && l->value)
     {
-      __ESBMC_thread_key_destructors[i](l->value);
+      __ESBMC_thread_key_destructors[i]((void *)l->value);
```

Comment:
Why is `l->value` `const` in the first place?

Reply:
This is because of this: https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_setspecific.html. While pthread_key_create requires this https://linux.die.net/man/3/pthread_key_create.